testsuite/test_cim_obj.py

```diff
@@ -2343,7 +2345,230 @@ def test_CIMInstanceName_from_wbem_uri(
             assert isinstance(obj.host, type(exp_host))
 
 
+class Test_CIMInstanceName_from_instance(object):
+    # pylint: disable=too-few-public-methods
+    """
+    Test the static method that creates CIMInstance name from the combination
+    of a CIMClass and CIMInstance.
+    """
+    testcases = [
+        # Testcases for CIMInstanceName.from_instance().
+        # Each testcase has these items:
+        # * desc: Short testcase description.
+        # * cls_kwargs: Dict with Attributes from which test class is
+        #   constructed
+        # * inst_kwargs: Dict with Attributes from which test inst is
+        #   constructed
+        # * exp_result: Dict of all expected attributes of resulting
+        #     CIMInstanceName without host and namespace if expected to suceed.
+        #     Exception type, if expected to fail.
+        # * strict: Value of strict attribute on from_instance call
+        # * condition: Condition for testcase to run.
```

Comment:
I suggest to add the capability to express an expected exception type, and expected warnings.